compiler/rustc_expand/src/mbe/transcribe.rs

```diff
@@ -763,19 +764,29 @@ fn extract_ident<'a>(
             }
             return Ok(nt_ident.to_string());
         }
-        if let ParseNtResult::Tt(TokenTree::Token(
-            Token { kind: TokenKind::Ident(token_ident, is_raw), .. },
-            _,
-        )) = pnr
-        {
-            if let IdentIsRaw::Yes = is_raw {
-                return Err(dcx.struct_span_err(ident.span, RAW_IDENT_ERR));
+
+        if let ParseNtResult::Tt(TokenTree::Token(Token { kind, .. }, _)) = pnr {
+            if let TokenKind::Ident(symbol, is_raw) = kind {
+                if let IdentIsRaw::Yes = is_raw {
+                    return Err(dcx.struct_span_err(ident.span, RAW_IDENT_ERR));
+                }
+                return Ok(symbol.to_string());
```

Comment:
By the way, why are we returning a `String` here and not a `Symbol`?
It looks like `transcribe_metavar_expr` can work without allocations at all if we don't allocate here prematurely.

Reply:
Not sure, a lot has changed since the initial draft.

The function is now returning `Symbol`